clients/include/rocblas_arguments.hpp

```diff
@@ -205,6 +207,12 @@ struct Arguments
         str << x;
     }
 
+    // Float16 output
+    static void print_value(std::ostream& str, rocblas_half x)
+    {
+        str << float(x);
```

Comment:
Yes can do that, but keeping the half in a slightly different format might be okay as well.

Reply:
> Should this be `print_value(str, float(x));` (or something similar) instead? Or maybe the conversion doesn't work as I'm expecting.
> Yes can do that, but keeping the half in a slightly different format might be okay as well.

That is intentional.

We do not want to pass `rocblas_half` to `operator<<`, so we upconvert it to `float` and then output it. We are assuming that converting `rocblas_half` to `float` is more available, or less ambiguous, than directly outputting `rocblas_half` to a stream is. There have been cases in the past where `rocblas_half` outputted to a stream caused ambiguous overloads with Tensile or other libraries defining similar functions.

Yes, you can be recursive, and call `print_value` after the conversion to `float`, but it doesn't really help here.

We do not want `rocblas_half` to change the output format. Numbers are printed as in C without respect to their type, which is expressed separately in `Arguments` (such as `a_type`, `b_type`, `c_type`, `d_type`, `compute_type`).

Please do not change core IO routines in rocBLAS until I come back.

There are scripts which depend on the exact output format, and the YAML parser depends on a particular format.

Reply:
Sure can wait for you to return.   Not sure why the half wasn't covered before but it was ambiguous so had to add this helper to Arguments, all other changes are in the new class.

Reply:
> Sure can wait for you to return. Not sure why the half wasn't covered before but it was ambiguous so had to add this helper to Arguments, all other changes are in the new class.

Oh, so this wasn't already there?  (Oh, I see it is marked green as an addition.) I thought it already was there, for the ambiguity reasons stated. We've run into this before, so I'm surprised we didn't already handle it.

Reply:
Looking at the original code, Daine is correct, it should be `print_value(str, float(x));` so that it formats the FP value correctly for YAML syntax, such as `.nan` and `.inf`.

To be more explicit, it should be written `print_value(str, double(x));` so that it definitely goes to the `double` overload.

There maybe other places this change is needed.